### PR TITLE
Issue #8501: Records support check update for BooleanExpressionComplexity

### PR DESCRIPTION
**Pull request by nmancus1**
Issue #8501: Records support check update for BooleanExpressionComplexityCheck

Diff Regression projects: https://gist.githubusercontent.com/nmancus1/e3988f8092d919b5cbe70f5359c4d9e8/raw/a0f6f5860f8025c19868061dee6e0e40960b992f/projects-to-test-on.properties

Diff Regression config: https://gist.githubusercontent.com/nmancus1/4b2c936ce7126095e94192bd20b74f20/raw/48cf09835dfbac3b67aa7414b084dc8f2a5e3594/BooleanExpressionComplexityCheck.xml

### CODE REVIEW

**Comment by nmancus1** [comment]
Github, generate report

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/metrics/booleanexpressioncomplexity/InputBooleanExpressionComplexityRecordsAndCompactCtors.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        private boolean myBool() {
+           return  (a & b) ^ (a || b) | a; // violation
+        }
+    }
+
+    record MyRecord2(String myString, boolean a, boolean b) {
+
+        // in compact ctor
+        public MyRecord2{
+            boolean d = (a & b) ^ (a || b) | a; // should be violation
```
Comment (pbludov) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/metrics/booleanexpressioncomplexity/InputBooleanExpressionComplexityRecordsAndCompactCtors.java:21]: Please remove `should be`

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/metrics/booleanexpressioncomplexity/InputBooleanExpressionComplexityRecordsAndCompactCtors.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        }
+    }
+
+    record MyRecord4(int y) {
+        private record MyRecord5(int z) {
+            // in nested record in compact ctor
+            public MyRecord5{
+                boolean b = false;
+                boolean a = true;
+                boolean d = (a & b) ^ (a || b) | a; // should be violation
```
Comment (pbludov) [inline src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/metrics/booleanexpressioncomplexity/InputBooleanExpressionComplexityRecordsAndCompactCtors.java:42]: Please remove `should be`

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/metrics/booleanexpressioncomplexity/InputBooleanExpressionComplexityRecordsAndCompactCtors.java (hunk 1 above)
Comment (nmancus1) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/metrics/booleanexpressioncomplexity/InputBooleanExpressionComplexityRecordsAndCompactCtors.java]: Fixed.

File: src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/metrics/booleanexpressioncomplexity/InputBooleanExpressionComplexityRecordsAndCompactCtors.java (hunk 2 above)
Comment (nmancus1) [reply src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/metrics/booleanexpressioncomplexity/InputBooleanExpressionComplexityRecordsAndCompactCtors.java]: Fixed.

**Pull request merged**